src/Propel/Runtime/DataFetcher/AbstractDataFetcher.php

```diff
@@ -49,6 +49,7 @@ public function getDataObject()
      */
     public function fetchColumn($index = null)
     {
+        /** @var array|null $next */
```

Comment:
I think `fetch()` returns false if there are no more columns to fetch

Reply:
Yeah, we could check on this. I just thought in this context it seems impossible (theoretical problem).

Reply:
I think the check happens in the next line:
```php
        if (!$next) {
            return null;
        }
```

Reply:
array|bool|null

The bool true is the problem here
And instead of just checking on this, I simplified to the expected outcome as docblock.